### ISSUE TITLE
Allow itest_service to not be started at "boot"

### ISSUE THREAD

**Issue by darkrift** [dzbarsky/rules_itest#38]
The use case is for testing cron jobs. The ask is to not start a specific itest_service to allow the service_test to exclusively start it through svcctl.

This allows to properly setup the environment from the service_test itself before the "job" starts and then assert on the execution.

**Comment by dzbarsky** [comment]
Yep this has been on my wishlist but not implemented yet. One thing to think through is the behavior when a non-deferred service depends on a deferred one (and how deps work here in general, I think we may have punted on that for the Start rpc). You interested in giving it a shot?

**Comment by darkrift** [comment]
Non-deferred service depending on deferred services doesn't make sense.

We could probably even do that in Starlark at analysis time.

As for deferred services depending on other deferred services, this could be validated when calling the /start endpoint and validate that all dependencies are currently running.

Wdyt ?

**Comment by dzbarsky** [comment]
Sounds good to me!
